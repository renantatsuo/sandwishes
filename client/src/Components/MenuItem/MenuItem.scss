.menu_item {
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 100px;
  width: 540px;
  padding: 10px;
  margin-bottom: 15px;

  background: #ffffff;
  border: none;
  box-shadow: 3px 3px 5px rgba(0, 0, 0, 0.05);
  cursor: pointer;
  transition: background ease 300ms;
  text-decoration: none;
  color: #444444;

  &:hover,
  &:focus {
    background: #f1f2f3;
    outline: none;
  }

  .menu_item__photo img {
    height: 80px;
    width: auto;
  }

  .menu_item__info {
    flex: 1;
    margin: 0 20px;
    text-align: left;
    font-size: 14px;

    > div {
      margin: 5px;
    }

    .menu_item__info--title {
      font-weight: bold;
    }

    .menu_item__info--desc {
      color: #999999;
    }
  }

  .menu_item__cart svg {
    width: 32px;
    height: 32px;
    fill: #e73737;
    margin: 10px;
  }
}
