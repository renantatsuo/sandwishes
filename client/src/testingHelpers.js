import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'

export const renderWithRouter = (ui) => ({
  ...render(<Router>{ui}</Router>)
})
