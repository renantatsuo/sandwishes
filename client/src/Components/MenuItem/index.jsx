import React from 'react'
import { Link } from 'react-router-dom'
import { AddShoppingCartOutlined } from '@material-ui/icons'

import { convertFloatToMoney } from '../../helpers'
import './MenuItem.scss'

const MenuItem = ({ desc, photo, price, title, to }) => (
  <Link to={to} className="menu_item">
    <div className="menu_item__photo">
      <img src={photo} alt={title} />
    </div>
    <div className="menu_item__info">
      <div className="menu_item__info--title">{title}</div>
      <div className="menu_item__info--desc">{desc}</div>
      <div className="menu_item__info--price">
        R$ {convertFloatToMoney(price)}
      </div>
    </div>
    <div className="menu_item__cart">
      <AddShoppingCartOutlined />
    </div>
  </Link>
)

export default MenuItem
