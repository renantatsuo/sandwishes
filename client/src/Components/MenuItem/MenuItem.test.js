import React from 'react'
import { renderWithRouter } from '../../testingHelpers'
import MenuItem from './index'

describe('Test MenuItem component', () => {
  it('Should render with no errors', () => {
    const { container } = renderWithRouter(<MenuItem to="/" />)

    expect(
      container.querySelector('.menu_item__info--price').textContent
    ).toEqual('R$ 0,00')
    expect(container).toMatchSnapshot()
  })
})
