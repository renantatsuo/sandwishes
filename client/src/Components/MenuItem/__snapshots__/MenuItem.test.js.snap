// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MenuItem component Should render with no errors 1`] = `ShallowWrapper {}`;
